File: dotnet/Operon.Sdk/Models/SessionInfo.cs
using System;
using System.Collections.Generic;

namespace Operon.Sdk.Models;

/// <summary>
/// Session metadata derived from PAT validation and token claims.
/// </summary>
public sealed class SessionInfo
{
    public string UserId { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string CustomerId { get; set; } = string.Empty;
    public IReadOnlyList<string> Roles { get; set; } = Array.Empty<string>();
    public IReadOnlyDictionary<string, object?> FeatureFlags { get; set; } = new Dictionary<string, object?>();
    public string WorkstreamId { get; set; } = string.Empty;
    public string ChannelId { get; set; } = string.Empty;
    public string WorkspaceId { get; set; } = string.Empty;
    public string ParticipantDid { get; set; } = string.Empty;
    public string ParticipantId { get; set; } = string.Empty;
    public string ClientId { get; set; } = string.Empty;
    public string SessionId { get; set; } = string.Empty;
    public DateTimeOffset? ExpiresAt { get; set; }
    public int ExpiresInSeconds { get; set; }
}

File: java/src/main/java/cloud/operon/sdk/SessionInfo.java
package cloud.operon.sdk;

import java.time.Instant;
import java.util.List;
import java.util.Map;

/**
 * Session metadata derived from PAT claims and validation endpoint payload.
 */
public record SessionInfo(
    String userId,
    String email,
    String name,
    String customerId,
    List<String> roles,
    Map<String, Object> featureFlags,
    String workstreamId,
    String workspaceId,
    String participantDid,
    String participantId,
    String clientId,
    String sessionId,
    Instant expiresAt,
    int expiresInSeconds
) {
}
